services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationControllerITest.kt

```diff
@@ -246,6 +247,26 @@ class ApplicationControllerITest(@Autowired override val mockMvc: MockMvc) : Con
         verify { hankeService.generateHankeWithApplication(applicationInput, USERNAME) }
     }
 
+    @Test
+    @WithMockUser(USERNAME)
+    fun `create with hanke generation calls validation fails with invalid data and returns 400`() {
+        val applicationInput =
+            AlluDataFactory.createApplication()
+                .withCustomer(
+                    customer =
+                        AlluDataFactory.createCompanyCustomer()
+                            .withContacts(
+                                AlluDataFactory.createContact(orderer = true),
+                                AlluDataFactory.createContact(orderer = true)
+                            )
+                )
+                .toCableReportWithoutHanke()
```

Comment:
There's an extension for changing the contacs of the customer.
```suggestion
            AlluDataFactory.createApplication()
                .withCustomerContacts(
                    AlluDataFactory.createContact(orderer = true),
                    AlluDataFactory.createContact(orderer = true)
                )
                .toCableReportWithoutHanke()
```